src/frontend/src/optimizer/plan_node/logical_scan.rs

```diff
@@ -416,6 +461,7 @@ impl PredicatePushdown for LogicalScan {
             self.output_col_idx().iter().map(|i| Some(*i)).collect(),
             self.table_desc().columns.len(),
         ));
+        let predicate = ConditionRewriter::rewrite(predicate);
```

Comment:
I think it is still too special to only apply this rule for table scan. This rule looks like a subset of `LogicalFilterExpressionSimplifyRule`. Can we enable `LogicalFilterExpressionSimplifyRule` for all filters, to achieve the same result?

Reply:
> Can we enable LogicalFilterExpressionSimplifyRule for all filters, to achieve the same result?

One problem is the `predicate` to rewrite is embedded with `LogicalScan` - that will not be `rewrite` from root, if I remember correctly.

But yes, I'll try find a more general solution to this.

Reply:
IIRC, Only `rewrite_exprs_recursive` will rewrite expr recursively. It is safe to use `rewrite_exprs` to simplify filter.